### PR TITLE
Add Sell All and Lock to the Underground

### LINKED ISSUE #1627: [SUGGESTION] "Sell All" button in the Underground for treasures
Currently, the Underground has a list of all your treasures in a tab, but it can be slow and annoying to have to go through the list of items to sell them if you have a long list of treasures you have obtained.

**Current Behavior**
Selling all your treasures requires repeatedly pressing "Sell" on every single type of item owned.

**Improved Behavior**
A "Sell All" button could be added to this tab to allow the player to sell every treasure they own.

**Reason for change**
This is surely a relatively simple addition that would improve quality of life in the Underground, especially for newer players that are nowhere near obtaining enough items to even consider the daily deals.

**Detailed Description**
This button could be added to the middle of the bottom part of the tab, between the Total Value display and the Close button, as I have placed it below, or appear in a space between the end of the list and this area, only being visible when an item can be sold. I assume it would also move Fossils to the Day Care as well when clicked.

**Screenshots:**
tempsnip

### PR DESCRIPTION
**Pull request by JapuDCret**
## What
closes #1627

Adds a sell all button, to sell everything at once.
Adds a "lock"-feature for treasures, so one can exclude them from the "Sell All"-feature.

## Why

I think the suggestion by [TailGlowVM](https://github.com/TailGlowVM) was actually pretty good and I wanted to learn how the project is set up.

## How tested

I tested it with my own save, therefore also checking if the import/version upgrade works.

## Examples

* Sell dialog, after clicking "Sell all treasures"
  issue-1627_sell-all-btn_sell-most-things
* Warn message, when you try to sell an item, that is locked.
  issue-1627_sell-all-btn_trying-to-sell-locked-item
* Warn message, when you try to click "Sell all", but have no compatible items (or 0 of them).
  issue-1627_sell-all-btn_nothing-to-sell

### CODE REVIEW

**Comment by RedSparr0w** [comment]
I haven't had a chance to test this yet, but could we move the 🔒 to the left hand side of the button, and change it to a 🔓 while the item is locked?

File: src/scripts/Player.ts [hunk 1]
```diff
@@ -71,7 +71,8 @@ class Player {
         // TODO(@Isha) move to underground classes.
         const mineInventory = (savedPlayer.mineInventory || [])
             // TODO: Convert this to object spread after we're on TS modules
-            .map((v) => Object.assign({}, v, { amount: ko.observable(v.amount) }));
+            .map((v) => Object.assign({}, v, { amount: ko.observable(v.amount) }))
+            .map((v) => Object.assign({}, v, { sellLocked: ko.observable(v.sellLocked) }));
```
Comment (RedSparr0w) [inline src/scripts/Player.ts:75]: I haven't tested this, but I think it should work without mapping twice?
```suggestion
            .map((v) => Object.assign({}, v, {
                amount: ko.observable(v.amount),
                sellLocked: ko.observable(v.sellLocked),
            }));
```

**Comment by JapuDCret** [comment]
I think I adressed all your points.
I also removed the "Lock"-btn from Fossils
firefox_3wRQNCa4lv

**Review by RedSparr0w — APPROVED** [review]
Looks good, thanks for the PR! 👍

**Pull request merged**